packages/playwright-test/src/plugins/vcsPlugin.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { PlaywrightTestConfig, TestPlugin } from '../types';
+import { createGuid } from 'playwright-core/lib/utils';
+import { spawnAsync } from 'playwright-core/lib/utils/spawnAsync';
+
+const GIT_OPERATIONS_TIMEOUT_MS = 1500;
+
+export const vcs = (options?: VCSPluginOptions): TestPlugin => {
+  return {
+    name: 'playwright-vcs-plugin',
+
+    configure: async (config: PlaywrightTestConfig, configDir: string) => {
+      options = options || {};
+      let revision: Partial<Revision> | undefined = options.revision;
+      if (shouldRunGit(options.revision, options.mode)) {
```

Comment:
Without this, plugin does not add much value. Let's make it git-only and allow room for non-git plugins. Otherwise we claim vcs and fall short.